Shared/Samples/Display map from portal item/DisplayMapFromPortalItemView.swift

```diff
@@ -0,0 +1,122 @@
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+struct DisplayMapFromPortalItemView: View {
+    /// The current portal item map.
+    @State private var currentMap = mapOptions.first!
+    
+    /// A map to display on the screen.
+    @State private var map = Map(item: mapOptions.first!.portalItem)
+    
+    /// A Boolean indicating whether to show the map options sheet.
+    @State private var isShowingMapOptions = false
+    
+    /// The different map options the user can choose from.
+    private static let mapOptions = [
+        PortalItemMap(title: "Terrestrial Ecosystems of the World",
+                      thumbnailImage: "TerrestrialEcosystemsThumbnail",
+                      portalID: .terrestrialEcosystems),
+        PortalItemMap(title: "Recent Hurricanes, Cyclones and Typhoons",
+                      thumbnailImage: "HurricanesCyclonesTyphoonsThumbnail",
+                      portalID: .hurricanesCyclonesTyphoons),
+        PortalItemMap(title: "Geology of United States",
+                      thumbnailImage: "USGeologyThumbnail",
+                      portalID: .usGeology)
+    ]
+    
+    var body: some View {
+        MapView(map: map)
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Button("Maps") {
+                        isShowingMapOptions = true
+                    }
+                }
+            }
+            .sheet(isPresented: $isShowingMapOptions, detents: [.medium], dragIndicatorVisibility: .visible) {
+                List {
+                    // Map options list.
+                    ForEach(DisplayMapFromPortalItemView.mapOptions) { mapOption in
+                        Button {
+                            currentMap = mapOption
+                            map = Map(item: currentMap.portalItem)
+                            isShowingMapOptions = false
+                        } label: {
+                            ZStack {
+                                HStack {
+                                    Image(uiImage: mapOption.thumbnailImage)
+                                    Text(mapOption.title)
+                                    Spacer()
+                                    Image(systemName: "checkmark")
+                                        .opacity(currentMap == mapOption ? 1 : 0)
+                                }
+                            }
+                        }
+                        .foregroundColor(.black)
+                    }
+                }
+            }
+    }
+}
+
+private extension DisplayMapFromPortalItemView {
+    /// A model for the maps the user may toggle between.
+    struct PortalItemMap: Equatable, Identifiable {
+        /// An id to conform to `Identifiable`.
+        let id = UUID()
+        
+        /// The text title of the map.
+        let title: String
+        
+        /// The UI thumbnail Image associated with the map.
+        let thumbnailImage: UIImage
```

Comment:
I think it is OK not to include the thumbnail image, as it is not mentioned in the common design.

If you want though, maybe explore https://developers.arcgis.com/swift/api-reference/documentation/arcgis/item/thumbnail and load the images asynchronously, instead of hardcoding them in the bundle.